### PR TITLE
Changed gcUnknownOutboundReferences to error and added test to validate GC's  refresh from snapshot is correct

### PR DESCRIPTION
**Pull request by agarwal-navin**
Updated "gcUnknownOutboundReferences" to be logged as an error - It should have been an error to begin with since its not expected.
Added couple of tests that show the bug where GC state is not updated correctly in GarbageCollector on refreshing state from a snapshot. Basically, the gc data from last run and tombstone state should also be updated when refereshing state from a snapshot. Otherwise, there are couple of things that can go wrong:
- The unreferenced timestamp of nodes is not updated in certain scenarios when it goes from unreferenced -> referenced -> unreferenced between summaries. Added a test for this scenario.
- "gcUnknownOutboundReferences" error can be logged in certain scenarios. Added a test for this scenario.

These tests repro the issues in bugs [AB#2940](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2940) and [AB#2945](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2945)

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts
Comment (tyler-cai-microsoft) [inline packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts:292]: Should we add a temporary `assertNotTrue` checking to see that `gcUnknownOutboundReferences` isn't even logged? I assume it's logged as a generic event

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM, I assuming you're merging this after we merge the refresh latest snapshot https://github.com/microsoft/FluidFramework/pull/13427/files

File: packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts
Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts:292]: It's not logged and that's why its not part of the "ifExpects" block. It will start showing up after this PR is merged which is hiding the bug in this scenario - https://github.com/microsoft/FluidFramework/pull/13361

Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts:292]: I will update this PR once #13361 is merged.

Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts:292]: Oh yeah, that's right, why do we still see `fluid:telemetry:Summarizer:Running:gcUnknownOutboundReferences` today then?

**Pull request merged**
